astro/src/content/quickstarts/quickstart-javascript-nextjs-web.mdx

```diff
@@ -0,0 +1,232 @@
+
+```shell
+npm install next-auth
+```
+
+Copy environment variables from our complete application example.
+
+```shell
+cp ../complete-application/.env.example .env.local
+```
+
+Also copy an image file into a new directory within `public` called `img`.
+
+```shell
+mkdir ./public/img && cp ../complete-application/public/img/money.jpg ./public/img/money.jpg
+```
+
+As you will be recreating all of the files in our app directory, please delete all files within `/src/app`.
+
+```shell
+rm -rf ./src/app && mkdir ./src/app
+```
+
+## Authentication
+
+Next.js 13.2 introduced <a href="https://nextjs.org/docs/app/building-your-application/routing/router-handlers" target="_blank">Route Handlers</a>, which are the preferred way to handle REST-like requests. In the `Changebank` application you can configure NextAuth.js FusionAuth's provider in a new route handler by creating a file within `src/app/api/auth/[...nextauth]/route.ts`.
+
+On first load of Next.js this file will make sure that you have all of the correct environment variables. The variables are then exported in an object called `authOptions` which can be imported on the server when you need to get our session using `getServerSession`.
+
+The `FusionAuthProvider` is then provided to `NextAuth` as a provider for any `GET` or `POST` commands that are sent to the `/api/auth/*` route.
+
+Create a new file named `src/app/api/auth/[...nextauth]/route.ts` and copy the following CSS for the ChangeBank application.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/api/auth/[...nextauth]/route.ts"}
+  lang="ts" />
+
+## App Customization
+
+### Styles
+
+Create a new file named `src/app/globals.css` copy the below CSS for the ChangeBank application.
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/app/globals.css"}
+  lang="ts" />
+
+### Login Button
+
+Create a new file in `src/components/login-button.tsx` that will be used for a button component. Our login button will only be used on the client side so make sure to add `use client` at the top of this file.
+For this button you can use the `signIn` and `signOut` functions from `next-auth/react`. By passing in the session from our pages you can determine if the `Log in` or `Log out` should be shown.
+
+Create a new file named `src/components/login-button.tsx` copy the below code for the ChangeBank application.
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/components/LoginButton.tsx"}
+  lang="ts" />
+
+### Layout
+
+<Aside type="tip">
+If this is your first time using Next.js's App Router, you should read through <a href="https://nextjs.org/docs/app/building-your-application/routing" target="_blank">Routing Fundamentals</a>.
+</Aside>
+
+Below you will find the full code for the [Root Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required)
```

Comment:
```suggestion
Below you will find the full code for the [Root Layout](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required).
```

I like to copy and paste the entire quickstart into Google docs and run a spell and grammar check on it. That catches stuff like this.